### PR TITLE
chore(release): cut v1.1.0-rc.1

### PR DESCRIPTION
**Pull request by schristoff**
- [ x] I have cherry-picked any changes [labelled](https://github.com/openservicemesh/osm/labels) `needs-cherry-pick-vX.Y.Z`
- [ x] I have made all of the following version and patch updates:
  1. Updated the container image tag in charts/osm/values.yaml
  2. Updated the chart **and** app version in charts/osm/Chart.yaml
  3. Updated the image tags used in the automated demo
  4. Regenerated the Helm chart README.md

- [x ] I have checked that the base branch for this PR is correct as defined by the release guide

Is this a release candidate?

### CODE REVIEW

File: .github/workflows/nightly-images.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
     env:
       DOCKER_USER: ${{ secrets.RELEASE_DOCKER_USER }}
       DOCKER_PASS: ${{ secrets.RELEASE_DOCKER_PASS }}
-      CTR_TAG: latest-main
+      CTR_TAG: v1.1.0
```
Comment (nojnhuh) [inline .github/workflows/nightly-images.yml:13]: All of the CI config should stay the same

File: charts/osm/values.yaml [hunk 2]
```diff
@@ -12,7 +12,7 @@ osm:
     # -- Container image pull policy for control plane containers
     pullPolicy: IfNotPresent
     # -- Container image tag for control plane images
-    tag: "latest-main"
+    tag: "v1.1.0"
```
Comment (nojnhuh) [inline charts/osm/values.yaml:15]: This should be blank so the digests take precedent by default.
```suggestion
    tag: ""
```

File: tests/e2e/e2e_upgrade_test.go [hunk 3]
```diff
@@ -49,7 +49,7 @@ var _ = OSMDescribe("Upgrade from latest",
 			// release branches, it should specify the most recent patch of the
 			// previous minor release. e.g. on the release-v1.0 branch, this
 			// should be "0.11".
-			i.Version = ">0.0.0-0"
+			i.Version = ">0.1.0-0"
```
Comment (nojnhuh) [inline tests/e2e/e2e_upgrade_test.go:52]: Like the previous release branch: https://github.com/openservicemesh/osm/blob/26886c4f16ae5cc92ddfe4c62c87cd82f1cd965c/tests/e2e/e2e_upgrade_test.go#L51
```suggestion
			i.Version = "1.0"
```

File: cmd/osm-bootstrap/osm-bootstrap_test.go [hunk 4]
```diff
@@ -43,7 +43,7 @@ var testPresetMeshConfigMap *corev1.ConfigMap = &corev1.ConfigMap{
 	"logLevel": "error",
 	"maxDataPlaneConnections": 0,
 	"envoyImage": "envoyproxy/envoy-alpine:v1.19.3@sha256:874e699857e023d9234b10ffc5af39ccfc9011feab89638e56ac4042ecd4b0f3",
-	"initContainerImage": "openservicemesh/init:latest-main",
+	"initContainerImage": "openservicemesh/init:v1.1.0",
```
Comment (nojnhuh) [inline cmd/osm-bootstrap/osm-bootstrap_test.go:46]: This test will still pass as-is, so we don't need to change this. If the test passes without this line at all, I'd also be in favor of removing it.

**Review by nojnhuh — CHANGES_REQUESTED** [review]

File: charts/osm/Chart.yaml [hunk 5]
```diff
@@ -18,7 +18,7 @@ version: 1.0.0
 
 # This is the version number of the application being deployed. This version number should be
 # incremented each time you make changes to the application.
-appVersion: latest-main
+appVersion: v1.1.0
```
Comment (nojnhuh) [inline charts/osm/Chart.yaml:21]: Also update `version` here too.
```suggestion
appVersion: v1.1.0-rc.1
```

Comment (jaellio) [inline charts/osm/Chart.yaml:21]: Are we cutting an RC first? v1.1.0-rc.1

File: charts/osm/Chart.yaml [hunk 6]
```diff
@@ -18,7 +18,7 @@ version: 1.0.0
```
Comment (jaellio) [inline charts/osm/Chart.yaml:18]: The `version` field should also be updated

File: charts/osm/README.md [hunk 7]
```diff
@@ -1,6 +1,6 @@
 # Open Service Mesh Helm Chart
 
-![Version: 1.0.0](https://img.shields.io/badge/Version-1.0.0-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: latest-main](https://img.shields.io/badge/AppVersion-latest--main-informational?style=flat-square)
+![Version: 1.0.0](https://img.shields.io/badge/Version-1.0.0-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: v1.1.0](https://img.shields.io/badge/AppVersion-v1.1.0-informational?style=flat-square)
```
Comment (jaellio) [inline charts/osm/README.md]:
```suggestion
![Version: 1.1.0-rc.1](https://img.shields.io/badge/Version-1.1.0--rc.1-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: v1.1.0-rc.1](https://img.shields.io/badge/AppVersion-v1.1.0--rc.1-informational?style=flat-square)
```

**Review by nojnhuh — APPROVED** [review]

**Review by jaellio — APPROVED** [review]

**Pull request merged**